src/psyclone/tests/psyir/frontend/fparser2_test.py

```diff
@@ -423,6 +427,125 @@ def test_process_declarations_intent(f2008_parser):
         Symbol.Access.READWRITE
 
 
+def test_process_declarations_kind_new_param(f2008_parser):
+    ''' Test that process_declarations handles variables declared with
+    an explicit KIND parameter that has not already been declared. '''
+    from fparser.two.Fortran2003 import Specification_Part
+    fake_parent = KernelSchedule("dummy_schedule")
```

Comment:
All new tests use the same processor, fake_parent, reader, fparser2spec but with different input Fortran. These should be made into a utility function.

Reply:
Oh yes, that's nice.